### PR TITLE
Ncl 1157 rsql

### PR DESCRIPTION
**Pull request by vibe13**
@alexcreasy Initial implementation for archived and running builds.
Examples of how it will work:

    by userId : pnc-rest/rest/builds?q=(user.id==1)
    by username: pnc-rest/rest/builds?q=(user.username==demo-user) / pnc-rest/rest/builds?q=(user.username=like=demo-user)
    by buildconfigurationid: pnc-rest/rest/builds?q=buildConfigurationAudited.idRev.id==1
    by buildconfigurationname: pnc-rest/rest/builds?q=buildConfigurationAudited.name==jboss-modules-1.5.0

NOTE: some endpoints are duplicated to verify correct functioning, will be polished later

### CODE REVIEW

File: rest/src/main/java/org/jboss/pnc/rest/endpoint/BuildConfigurationEndpoint.java [hunk 1]
```diff
@@ -507,4 +507,22 @@ public Response getLatestBuildRecord(@ApiParam(value = "Build configuration id",
         return this.fromSingleton(buildRecordProvider.getLatestBuildRecord(id));
     }
 
+    @ApiOperation(value = "Get all BuildRecords (running and archived) associated with this Build Configuration, returns empty list if no build records are found")
```
Comment (vibe13) [inline rest/src/main/java/org/jboss/pnc/rest/endpoint/BuildConfigurationEndpoint.java]: To be removed after testing, will be available via `pnc-rest/rest/builds?q=buildConfigurationAudited.idRev.id==1`

File: rest/src/main/java/org/jboss/pnc/rest/endpoint/UserEndpoint.java [hunk 2]
```diff
@@ -193,4 +193,22 @@ public Response update(@ApiParam(value = "User id", required = true) @PathParam(
        return super.update(id, userRest);
     }
 
+    @ApiOperation(value = "Get all BuildRecords (running and archived) triggered by this User, returns empty list if no build records are found")
```
Comment (vibe13) [inline rest/src/main/java/org/jboss/pnc/rest/endpoint/UserEndpoint.java]: To be removed after testing, will be available via `pnc-rest/rest/builds?q=(user.id==1)`

**Comment by alexcreasy** [comment]
+1 for merging when CI passes

**Comment by vibe13** [comment]
@alexcreasy forgot some commits :)

**Comment by vibe13** [comment]
retest this please

**Comment by vibe13** [comment]
retest this please

**Comment by vibe13** [comment]
retest this please

**Comment by vibe13** [comment]
retest this please

**Comment by vibe13** [comment]
retest this please

File: rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildExecutionConfigurationRest.java [hunk 3]
```diff
@@ -95,6 +116,20 @@ public BuildExecutionConfiguration toBuildExecutionConfiguration() {
         );
     }
 
+    public BuildConfigurationAudited toBuildConfigurationAudited() {
+        BuildConfigurationAudited buildConfigAudited = new BuildConfigurationAudited();
```
Comment (matejonnet) [inline rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildExecutionConfigurationRest.java]: This object was not meant to be directly mapped to DB entities and a requirement is to make rest-model independent of DB model.

**Comment by vibe13** [comment]
@matejonnet merged and reverted

File: rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildRecordRest.java [hunk 4]
```diff
@@ -128,9 +140,20 @@ public BuildRecordRest(BuildRecord buildRecord) {
         this.distributedMilestoneBuildRecordSetIds = nullableStreamOf(buildRecord.getBuildRecordSets())
                 .filter(buildRecordSet -> buildRecordSet.getDistributedInProductMilestone() != null)
                 .map(buildRecordSet -> buildRecordSet.getId()).collect(Collectors.toSet());
+
+        // To prevent lazy loading of collections, go through the transformation chain
```
Comment (matejonnet) [inline rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildRecordRest.java]: Can we hide complex ORM code in DatastoreAdapter? *Rest DTOs should be as simple as possible and deal mostly with (de)serialization.

File: rest/src/main/java/org/jboss/pnc/rest/endpoint/UserEndpoint.java (hunk 2 above)
Comment (matejonnet) [inline rest/src/main/java/org/jboss/pnc/rest/endpoint/UserEndpoint.java]: Should we leave at least a TODO note?

File: rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildExecutionConfigurationRest.java (hunk 3 above)
Comment (vibe13) [inline rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildExecutionConfigurationRest.java]: @matejonnet  Ok I see your point, thanks. This conversion is done just to mimic the relation that the BuildTask has with the BuildConfiguration, in order to use the rsql query also with the in memory Builds (the running ones).What if I convert to BuildConfigurationAuditedRest? There is already a UserRest inside the BuildExecutionConfigurationRest object?

Comment (matejonnet) [inline rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildExecutionConfigurationRest.java]: I don't see it as a conversion as the two entities are not representing the same data. At the end the two entities might live on two different deployments (coordinator & executor).
Can't you work directly with this entity?

**Pull request merged**